doc/user/content/transform-data/patterns/rules-engine.md

```diff
@@ -0,0 +1,149 @@
+In our example, for each rule in a `bird_rules` dataset, we filter the `birds` dataset according to the rule.
+
+### Create Resources
+
+1. Create the `birds` table and insert some birds.
+    ```sql
+    CREATE TABLE birds (
+    id INT,
+    name VARCHAR(50),
+    wingspan_cm FLOAT,
+    colors jsonb
+    );
+
+    INSERT INTO birds (id, name, wingspan_cm, colors) VALUES
+    (1, 'Sparrow', 15.5, '["Brown"]'),
+    (2, 'Blue Jay', 20.2, '["Blue"]'),
+    (3, 'Cardinal', 22.1, '["Red"]'),
+    (4, 'Robin', 18.7, '["Red","Brown"]'),
+    (5, 'Hummingbird', 8.2, '["Green"]'),
+    (6, 'Penguin', 99.5, '["Black", "White"]'),
+    (7, 'Eagle', 200.8, '["Brown"]'),
+    (8, 'Owl', 105.3, '["Gray"]'),
+    (9, 'Flamingo', 150.6, '["Pink"]'),
+    (10, 'Pelican', 180.4, '["White"]');
+    ```
+1. Create the `bird_rules` table and create a few rules.
+    ```sql
+    CREATE TABLE bird_rules (
+    id INT,
+    starts_with CHAR(1),
+    wingspan_operator VARCHAR(3),
+    wingspan_cm FLOAT,
+    colors JSONB
+    );
+
+    INSERT INTO bird_rules (id, starts_with, wingspan_operator, wingspan_cm, colors)
+    VALUES
+    (1, 'P', 'GTE', 50.0, '["Blue"]'),
+    (2, 'P', 'LTE', 100.0, '["Black","White"]'),
+    (3, 'R', 'GTE', 20.0, '["Red"]');
+    ```
+    Each rule has a unique `id` and encodes filters on starting letter, wingspan, and color. For `wingspan_operator`, `'GTE'` means "greater than or equal" and `'LTE'` means "less than or equal". For more complicated rules with varying schemas, you might consider using the [`jsonb` type](/sql/types/jsonb) and adjust the logic in the upcoming `LATERAL` join to suit your needs.
+
+### Create the View
+
+Here is the view that will execute our bird rules:
+
+```sql
+CREATE VIEW birds_filtered AS
+SELECT r.id AS rule_id, b.name, b.colors, b.wingspan_cm
+FROM
+-- retrieve bird rules
+(SELECT id, starts_with, wingspan_operator, wingspan_cm, colors FROM bird_rules) r,
+-- for each bird rule, find the birds who satisfy it
+LATERAL (
+    SELECT *
+    FROM birds
+    WHERE r.starts_with = SUBSTRING(birds.name, 1, 1)
+        AND (
+            (r.wingspan_operator = 'GTE' AND birds.wingspan_cm >= r.wingspan_cm)
+            OR
+            (r.wingspan_operator = 'LTE' AND birds.wingspan_cm <= r.wingspan_cm)
+        )
+        AND r.colors <@ birds.colors
+) AS b;
+```
+
+### Subscribe to Changes
+
+1. Subscribe to the changes of `birds_filtered`.
+    ```sql
+    COPY(SUBSCRIBE birds_filtered) TO STDOUT;
+    ```
+    ```nofmt
+    mz_timestamp  | mz_diff | rule_id |   name   |      colors         | wingspan_cm
+    --------------|---------|---------|----------|---------------------|------------
+    1688673701670      1         2       Penguin     ["Black","White"]       99.5
+    ```
+    Notice that the majestic penguin satisfies a rule, and in fact it's rule 2. None of the other birds satisfy any of the rules.
```

Comment:
```suggestion
    Notice that the majestic penguin satisfies rule 2. None of the other birds satisfy either of the rules.
```

Reply:
There are 3 rules at this point, so I will keep "any". I will update the first sentence though.